### PR TITLE
thank you page track

### PR DESCRIPTION
**Pull request by MihaiCernat**

### CODE REVIEW

**Comment by MihaiCernat** [comment]
[P1 || [Segment - Thank you page] Add pageview and event to Shopify order status / thank you page](https://trello.com/c/NZOGOMs7/4059-p1-segment-thank-you-page-add-pageview-and-event-to-shopify-order-status-thank-you-page)

File: src/gaThankYouPageTracker/gaThankYouPageTracker.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+(function () {
+	const scriptSrc = document.currentScript.src;
+	const startIndex = scriptSrc.indexOf('webPropertId=');
```
Comment (TheGame2500) [inline src/gaThankYouPageTracker/gaThankYouPageTracker.ts:3]: @MihaiCernat I think there's a typo here. Should be `webPropertyId`

File: src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
+(function () {
+	const scriptSrc = document.currentScript.src;
+	const startIndexGa = scriptSrc.indexOf('webPropertId=');
```
Comment (TheGame2500) [inline src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts:3]: @MihaiCernat same typo here

File: src/gaThankYouPageTracker/gaThankYouPageTracker.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	const webPropertId = scriptSrc.subscting(startIndex + 14);
```
Comment (TheGame2500) [inline src/gaThankYouPageTracker/gaThankYouPageTracker.ts:4]: @MihaiCernat can we abstract getting the segmentProperty/webPropertyId in a function in helpers.ts?

**Review by TheGame2500 — CHANGES_REQUESTED** [review]
You also need to add a condition that this only runs if the current URL is the thank you page, otherwise this will run on all pages of the store

**Review by TheGame2500 — APPROVED** [review]

File: src/gaThankYouPageTracker/gaThankYouPageTracker.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+		// @ts-ignore
+		gtag('js', new Date());
+		// @ts-ignore
+		gtag('config', webPropertId);
+		// @ts-ignore
+		const transaction_total = window.Shopify.checkout.total_price;
+		// @ts-ignore
+		gtag('event', 'Checkout', {
+			event_category: 'Shopify (Littledata)',
+			event_label: 'Thank you',
```
Comment (edwardupton) [inline src/gaThankYouPageTracker/gaThankYouPageTracker.ts:30]: Should be 'Thank you page'

File: src/gaThankYouPageTracker/gaThankYouPageTracker.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+			value: transaction_total,
```
Comment (edwardupton) [inline src/gaThankYouPageTracker/gaThankYouPageTracker.ts:31]: Event values have to be integers, so round to nearest

File: src/gaThankYouPageTracker/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export const getWebProperty = (scriptSrc: string): string => {
```
Comment (edwardupton) [inline src/gaThankYouPageTracker/helpers.ts:1]: This isn't needed - you can use getProperties as a shared helper

File: src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+		window.dataLayer = window.dataLayer || [];
+		const stubFunction = function () {
+			dataLayer.push(arguments);
+		}; //eslint-disable-line
+		// @ts-ignore
+		window.gtag = window.gtag || stubFunction;
+		// @ts-ignore
+		gtag('js', new Date());
+		// @ts-ignore
+		gtag('config', webPropertyId);
```
Comment (edwardupton) [inline src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts:24]: No, you don't need to init gtag for Segment - Segment already does that

File: src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+		const analytics = (window.analytics = window.analytics || []);
```
Comment (edwardupton) [inline src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts:26]: There is already a helper for initializing Segment analyticsJS

File: src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+				};
+				analytics.SNIPPET_VERSION = '4.13.1';
+				analytics.load(segmentProperty);
+				analytics.page();
+			}
+
+		const checkout = window.Shopify.checkout;
+		const products = [];
+
+		let gaClientId;
```
Comment (edwardupton) [inline src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts:82]: Not needed - Segment handles this

File: src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+				order_id: checkout.order_id,
+				presentment_currency: checkout.presentment_currency,
+				presentment_total: checkout.total_price_set.presentment_money.amount,
+				products: products,
+				sent_from: 'Littledata app',
+				shipping: checkout.shipping_rate && checkout.shipping_rate.price,
+				tax: checkout.total_tax,
+				total: checkout.total_price,
+			},
+			integrations: {
```
Comment (edwardupton) [inline src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts:116]: Not needed - Segment handles this

File: src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+				if (gaCookie) {
+					const match = gaCookie.match(/(\d{2,11})\.(\d{2,11})/g);
+					gaClientId = match ? match[0] : '';
+				}
+			}
+		}
+
+		analytics.track('Thank you', {
+			properties: {
+				category: 'Shopify (Littledata)',
```
Comment (edwardupton) [inline src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts:102]: Not required

File: src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+				coupon: checkout.coupon,
+				currency: checkout.currency,
+				discount: checkout.discount,
+				email: checkout.email,
+				order_id: checkout.order_id,
+				presentment_currency: checkout.presentment_currency,
+				presentment_total: checkout.total_price_set.presentment_money.amount,
```
Comment (edwardupton) [inline src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts:109]: Do you need to check for existence of total_price_set and presentment_money?

File: src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+				products: products,
```
Comment (edwardupton) [inline src/segmentThankYouPageTracker/segmentThankYouPageTracker.ts:110]: You need to translate fields into segmentProduct format

**Review by edwardupton — CHANGES_REQUESTED** [review]

**Review by edwardupton — APPROVED** [review]

**Pull request merged**